src/draughts/Pawn.java

```diff
@@ -73,32 +72,76 @@ public boolean validateMove(Board board, Coordinates position) {
 
     public Pawn validateMoveWithCapture(Board board, Coordinates position) {
         Pawn[][] fields = board.getFields();
-        //todo: null point exception prawdopodobnie juz nie aktualne -> blad rozwiazany ale nalezy zweryfikowac
-//        error reason:
-//        Podaj współrzędne pionka, którym chcesz wykonać ruch, np. A1
-//        a10
-//        Podaj współrzędne pola, na które chcesz wykonać ruch, np. A1
-//                b10
-//        ERROR:
-//        Exception in thread "main" java.lang.NullPointerException: Cannot read field "color" because "fields[goalX][goalY]" is null
-//        at draughts.Pawn.validateMoveWithCapture(Pawn.java:75)
-//        at draughts.Game.tryToMakeMove(Game.java:175)
-//        at draughts.Game.checkStartingPosition(Game.java:152)
-//        at draughts.Game.playRound(Game.java:51)
-//        at draughts.Game.start(Game.java:31)
-//        at draughts.Main.main(Main.java:8)
+
         int startX = this.position.getX();
+
         int startY = this.position.getY();
         int goalX = position.getX();
         int goalY = position.getY();
-        Color startColor = this.color;
-
-        //TODO:sprawdz czy wybrane pole jest o dwa pola oraz czy jest puste po przekatnej dalej oraz czy miedzy nimi jest
-        // pionek przeciwnego koloru,jezeli tak to zwroc pionek do bicia, w przeciwnym razie zwroc null
-        //hint rozubodowana validacja z ValidateMove()
+        Pawn goalPawn = fields[goalX][goalY];
+
+        // if goal field is empty
+        if (goalPawn == null) {
+            // if my color is black
+            if (this.color.equals(black)) {
+                // if my goal move is 2 fields diagonally away
+                if ((goalX == startX - 2) && (goalY == startY - 2)) {
```

Comment:
I guess we can simplify that conditions just to one half (separate tree for black and white into one for both colors) if we substitute -2 with variable which will change between 2 and -2 depending on the color of the pawn